### ISSUE TITLE
[SpecMoE] P2-1: Router Logits 分歧检测可行性研究

### ISSUE THREAD

**Issue by Li-changwu** [Li-changwu/MoE-SD#35]
## 🎯 目标

验证是否可以通过 router logits 的层间分歧来预测 draft token 的拒绝。

## 📐 理论基础

对于 draft token t_j，如果 target model 在第 l 层的 router logits 与 draft model
预期的差异很大（KL散度高），则该 token 大概率会被拒绝。

Speculation Divergence Detector (SDD):
```
divergence(t_j, l) = KL(router_target^(l)(t_j) || router_draft^(l)(t_j))
if divergence > θ for N consecutive layers → early terminate t_j
```

### Step 1: 收集 Draft vs Target Router Logits
- 在 EAGLE-3 verify 过程中 hook 每层 router
- 对 K 个 draft token，记录:
  - target router logits (48 layers × 128 experts)
  - 对应的 acceptance/rejection 结果

### Step 2: 分析分歧信号
- 计算每层的 KL 散度 / L2 距离 / top-k overlap
- 画出: accepted tokens vs rejected tokens 的分歧分布
- 找到: 最早可区分的层 (layer l*)

### Step 3: 量化 SDD 指标
- Precision: 判定为 reject 的 token 中实际被 reject 的比例
- Recall: 实际被 reject 的 token 中被 SDD 检测到的比例
- 最优 (θ, N) 参数
- 平均检测层 l*（越早越好）

## 📦 输出

- `results/sdd_analysis/divergence_trace.jsonl`
- `results/sdd_analysis/sdd_precision_recall.csv`
- `results/sdd_analysis/optimal_layer_threshold.json`
- Plot: divergence distribution (accept vs reject, per layer)
- Plot: precision-recall curve vs threshold θ

## ✅ 验收标准

- [ ] precision >80%, recall >60% 则 SDD 可行
- [ ] 平均检测层 l* < L/2 = 24 则有显著收益
- [ ] 若不满足 → 关闭此研究方向，在 issue 中记录 negative result
